Sources/Datadog/Core/AutoInstrumentation/URLSessionSwizzler.swift

```diff
@@ -11,39 +11,35 @@ import Foundation
 /// Takes an URL and if it is to be intercepted, returns a TaskObserver and additional HTTP headers
 /// otherwise, returns nil
 internal typealias RequestInterceptor = (URL?) -> InterceptionResult?
-internal typealias InterceptionResult = (taskObserver: TaskObserver, httpHeaders: [String: String])
+internal struct InterceptionResult {
+    let taskObserver: TaskObserver
+    let httpHeaders: [String: String]
+}
 
 /// Block to be executed at task starting and completion by URLSessionSwizzler
 /// starting event is passed at task.resume()
 /// completed event is passed when task's completion handler is being executed
 internal typealias TaskObserver = (TaskObservationEvent) -> Void
-internal enum TaskObservationEvent: Equatable {
-    case starting
-    case completed
+internal enum TaskObservationEvent {
+    case starting(URLRequest?)
+    case completed(URLResponse?, Error?)
 }
 
 internal class URLSessionSwizzler {
     let dataTaskWithURL: DataTaskWithURL
     let dataTaskwithRequest: DataTaskWithRequest
-    let resume: Resume
+    /// resume must be shared among URLSessionSwizzler instances
+    /// to avoid swizzling task.resume() multiple times
+    static let resume = Resume()
```

Comment:
we took that decision verbally but the code allows multiple `URLSessionSwizzler` instances. 
either we can 
1. hope that the next developer will be aware of our verbal discussion 
2. or we can write code to prevent them from mistakes

Reply:
or

3. We can enforce it as part of the architecture that there's no reason to swizzle twice (i.e. by accepting `URLSessionSwizzler` instance in `TracingAutoInstrumentation.init()` and `RUMAutoInstrumentation.init()`).

which IMO would be a good and pragmatic trade-off between the code complexity and safety. Otherwise, we'll end up with self-defending unit tests and eventually 2 sort of error messages printed out to the user:
* _"XXX auto-instrumentation will not work"_,
* _"XXX auto-instrumentation will partially work"_.

IMO no reason to complicate this stuff.

Reply:
as we discussed before, we will have one `URLSessionSwizzler` but we will run `swizzler.swizzle(...)` twice.
that way we can make `TracingAutoInstrumentation` and `RUMAutoInstrumentation` separate and we won't need a new type such as `CombinedInterceptor` as Android does.

this simplifies the code

Reply:
Less code doesn't mean simpler code. It's definitely less complex if concerns are separated → combining interceptors as a distinct unit of work. It would be also healthier for unit tests as they won't be mixing everything up like now: interception logic + interceptors iteration in single test.

But at the end, it's the same business for now.